lips/lip-10.md

```diff
@@ -0,0 +1,231 @@
+  * *For example*: VASP can have notion of 'contact list' and only share information if the sender is in receiver's contacts
+* **Payment endpoint** to perform actual money transfer.
+
+Common considerations:
+
+* For both endpoints, all communications are signed via the same mechanism as for other off-chain communications, see the [Payloads section of lip-1](https://github.com/libra/lip/blob/master/lips/lip-1.mdx#payloads) for details. As such, all vasp-2-vasp communications are authenticated.
+
+* All Libra ID endpoints are nested under `pay.<version>` URI path:
+
+     `https://<hostname>:<port>/pay.<protocol_version>/<endpoint>`
+
+* All Libra ID endpoints use the same `base_url` and same `compliance_public_key` for authentication as other off-chain APIs.
+
+* All Libra ID endpoints are HTTP POST endpoints, they accept JSON-encoded requests and return JSON-encoded responses. Other encoding methods can be considered in the future, `Content-Type` and `Accept` HTTP headers can be used to define request/response formats.
+
+## Info endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/info
+```
+
+**Info endpoint** is used by the **Sender** to request information about the **Receiver** before submitting an actual payment.
+
+Info endpoint takes both sender and receiver Libra IDs as arguments. Receiver VASP might decide to share more(or less) information based on sender ID, for example if the wallet supports some kind of contact list, they might decide to share more information if the sender is in the contact list of the receiver.
+
+**Request:**
+```json
+{
+   "sender": "<User Object>",
+   "receiver_id": "<Receiver Libra ID>"
+}
+```
+
+| Field 	   | Type 	     | Required? 	| Description 	           |
+|-------	   |------	     |-----------	|-------------	           |
+|sender        |User Object  | Y            | Information about the sender |
+|receiver_id   |str          | Y            | Libra ID of the receiver |
+
+**Response:**
+```json
+{
+   "receiver": "<User Object>"
+}
+```
+
+| Field 	   | Type 	     | Required? 	| Description 	             |
+|-------	   |------	     |-----------	|-------------	             |
+|receiver      |User Object  | Y            | Information about the receiver |
+
+* Receiver Libra ID domain must belong to the Receiver VASP. Otherwise, Receiver VASP must return the `400` error code.
+* Sender Libra ID must have a Libra ID domain that belongs to Sender VASP. If this is not the case, the Receiver VASP should return the `400` error code 
+* Receiver VASP may optionally choose how much information to share based on the sender Libra ID. If Receiver VASP decides to not share data, it should return an empty `User Object` with only `libra_id` field set
+* If user is not found, the VASP should return error code `400`  
+
+## Pay endpoint
+
+```
+POST https://<hostname>:<port>/pay.<protocol_version>/pay
```

Comment:
This is a "proper" off-chain command and should be treated as such. We should aim to have the reasoning behind the URI scheme. (see above)

Reply:
Are you talking about adding `<LocalVASPAddress>/<RemoteVASPAddress>` part (agree with that) or something else?

Reply:
both the `<LocalVASPAddress>/<RemoteVASPAddress>` part and the `pay` resource name. 
In general, LIP-1 (and LIP-8 as its extension) uses a single resource URI (`/command`) as the main entry point to all off-chain communication. 
We should either keep it as is (and adapt `/command` notation here) or change it across all LIPs to be a logical representation of resources and actions. 
I also think that we should stick to the main principles of LIP-1 [CommandRequestObject](https://lip.libra.org/lip-1/#commandrequestobject) and [CommandResponseObject](https://lip.libra.org/lip-1/#commandresponseobject) as it makes things easier for developers to have a one consistent protocol scheme/format.

Reply:
We specifically chose not to use command request/response style here to keep it simpler

Reply:
I'd agree with @danprinz here.  If we are going to treat this as an action, then it should be using the same command structure as is used in LIP-1 and LIP-8.  That keeps us consistent and also means we don't need to re-state so much and make it seems sort of like LIP-1 but sort of its own thing